### ISSUE TITLE
Multiple scopes for `namespace foo.bar`

### ISSUE THREAD

**Issue by overlookmotel** [oxc-project/oxc#3471]
If I understand correctly, this:

```ts
namespace foo.bar {
  export const qux = 123;
}
```

is just shorthand for:

```ts
namespace foo {
  export namespace bar {
    export const qux = 123;
  }
}
```

Therefore, `namespace foo.bar` essentially represents 2 scopes - 1 for `foo`, and a 2nd for `bar`.

I propose to encode this in the AST by either:

1. Add `scope_id` field to the `TSModuleDeclarationBody::TSModuleDeclaration` variant
or
2. Remove `scope_id` field from `TSModuleBlock` and add it to `TSModuleDeclaration` instead.

Creating multiple scopes seems semantically correct to me, and would make transforming it to JS much simpler.

Just opening this issue because I'm pretty rubbish with TS, and want to check my reasoning before going ahead with it. Am I misunderstanding anything?

cc @Boshen @Dunqing

**Comment by Dunqing** [comment]
Sounds reasonable. I'm fine with that.

**Comment by overlookmotel** [comment]
Thanks. I'll do that, and then hopefully can get all the scopes in namespace transform correct.

**Comment by overlookmotel** [comment]
Implemented in #3488.
